src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -385,7 +385,7 @@ HashSet<ITypeSymbol> typeStack
 			if( expr.Kind() == SyntaxKind.NullLiteralExpression ) {
 				// This is perhaps a bit suspicious, because fields and
 				// properties have to be readonly, but it is safe...
-				return MutabilityInspectionResult.NotMutable();
+				return MutabilityInspectionResult.NotMutable( null );

```

Comment:
Maybe move this after `var exprType = typeInfo.Type ?? typeInfo.ConvertedType;`, yeah I think I need to change this, and have test around it.

Reply:
No that doesn't make sense. The typeInfo could just be say `string`, there isn't any location information associated with that.